functions/consumer/websocket-consumer/src/main/java/org/springframework/cloud/fn/consumer/websocket/WebsocketConsumerConfiguration.java

```diff
@@ -42,42 +43,51 @@
  * @author Oliver Moser
  * @author Gary Russell
  * @author Artem Bilan
+ * @author Chris Bono
  */
 @Configuration
 @EnableConfigurationProperties(WebsocketConsumerProperties.class)
 public class WebsocketConsumerConfiguration {
 
-
 	private static final Log logger = LogFactory.getLog(WebsocketConsumerConfiguration.class);
 
-	private final InMemoryTraceRepository websocketTraceRepository = new InMemoryTraceRepository();
-
 	@Value("${endpoints.websocketconsumertrace.enabled:false}")
 	private boolean traceEndpointEnabled;
 
+	@Autowired
+	private WebsocketConsumerServer websocketConsumerServer;
```

Comment:
This was another case of needing the configuration to create the thing that it needs to be injected w/. In this case it was the `server()` bean call in the `init` method. 

To get around this, I switched to wrapping the server aspects in a nested static configuration class so that those beans will be created by that configuration and then ultimately injected into the outer configuration.